src/libsyntax/codemap.rs

```diff
@@ -807,6 +807,70 @@ impl CodeMap {
                         hi.col.to_usize() + 1)).to_string()
     }
 
+    pub fn span_to_expanded_string(&self, sp: Span) -> String {
+        let src_str = self.span_to_string(sp);
+        let src_snip = self.span_to_snippet(sp).unwrap_or(
+            "Snippet unavailable (span malformed or source missing)".to_string());
```

Comment:
I feel like you shouldn't need this code and the same code inside the loop - maybe you can refactor the loop a little